casepropods/family_connect_subscription/plugin.py

```diff
@@ -78,6 +112,69 @@ def format_schedule(self, schedule):
             schedule['month_of_year'], schedule['day_of_week'])
         return prettify_cron(cron_schedule)
 
+    def cancel_subscriptions(self, subscription_ids):
+        params = {'active': False}
+        for subscription in subscription_ids:
+            try:
+                self.stage_based_messaging_api.update_subscription(
+                    subscription, params)
+            except HTTPServiceError:
+                return False
+        return True
+
+    def perform_action(self, type_, params):
+        if type_ == "cancel_subs":
+            subscription_ids = params.get("subscription_ids", [])
+            success = self.cancel_subscriptions(subscription_ids)
+            if not success:
+                return (False,
+                        {"message": "Failed to cancel some subscriptions"})
+            return (True, {"message": "cancelled all subscriptions"})
+
+        if type_ == "full_opt_out":
+            opted_out = False
+            subs_cancelled = False
+
+            # Create OptOut
+            contact_id = params["contact_id"]
+            contact = Contact.objects.get(pk=contact_id)
+            identity = contact.uuid
+
+            opt_out_url = settings.IDENTITY_API_ROOT + "api/v1/optout/"
+            identity_token = settings.IDENTITY_AUTH_TOKEN
+            headers = {
+                'Authorization': "Token " + identity_token,
+                'Content-Type': "application/json"
+            }
+
+            # Opt-outs have to have an address. This kinda sucks though
+            if contact.urns:
+                addr_type, address = contact.urns[0].split(':', 1)
```

Comment:
This is actually probably a bug in the identity store since the address is stored on the Opt-Out. Not ideal when they've asked to have all of their data removed